src/main/java/uk/gov/companieshouse/psc/extensions/api/controller/ValidationStatusControllerImpl.java

```diff
@@ -0,0 +1,76 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+import uk.gov.companieshouse.api.model.psc.PscIndividualFullRecordApi;
+import uk.gov.companieshouse.api.pscextensions.api.PscExtensionValidationStatusApi;
+import uk.gov.companieshouse.api.pscextensions.model.ValidationStatusResponse;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.companieshouse.psc.extensions.api.enumerations.PscType;
+import uk.gov.companieshouse.psc.extensions.api.exception.FilingResourceNotFoundException;
+import uk.gov.companieshouse.psc.extensions.api.exception.PscLookupServiceException;
+import uk.gov.companieshouse.psc.extensions.api.service.PscExtensionsService;
+import uk.gov.companieshouse.psc.extensions.api.service.PscLookupService;
+import uk.gov.companieshouse.psc.extensions.api.utils.LogMapHelper;
+
+
+import static uk.gov.companieshouse.psc.extensions.api.PscExtensionsApiApplication.APPLICATION_NAMESPACE;
+
+@RestController
+public class ValidationStatusControllerImpl implements PscExtensionValidationStatusApi {
+    private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAMESPACE);
+    private final PscExtensionsControllerImpl pscExtensionsController;
+    private final PscLookupService pscLookupService;
+    private final PscExtensionsService pscExtensionsService;
+
+    public ValidationStatusControllerImpl(final PscExtensionsControllerImpl pscExtensionsController,
+                                          final PscLookupService pscLookupService,
+                                          final PscExtensionsService pscExtensionsService) {
+        this.pscExtensionsController = pscExtensionsController;
+        this.pscLookupService = pscLookupService;
+        this.pscExtensionsService = pscExtensionsService;
+    }
+
+    @Override
+    public ResponseEntity<ValidationStatusResponse> _validate(@PathVariable("transactionId") final String transactionId,
+                                                              @PathVariable("filingResourceId") final String filingResourceId)
+    {
+        final HttpServletRequest request = ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()).getRequest();
+
+        final var logMap = LogMapHelper.createLogMap(transactionId);
+        logMap.put("path", request.getRequestURI());
+        logMap.put("method", request.getMethod());
+        LOGGER.debugRequest(request, "GET validation request", logMap);
+
+        final var pscExtensionOpt = pscExtensionsService.get(filingResourceId);
+        final var pscExtension = pscExtensionOpt.orElseThrow(() -> new FilingResourceNotFoundException(
+                String.format("PSC extension not found when generating filing for %s", filingResourceId)));
```

Comment:
reword message for clarity:

```suggestion
                String.format("PSC extension not found when validating filing for %s", filingResourceId)));
```